torchvision/transforms/v2/functional/_geometry.py

```diff
@@ -210,64 +210,73 @@ def resize_image(
         # error if other interpolation modes are used. This is documented.
         antialias = False
 
-    shape = image.shape
-    numel = image.numel()
-    num_channels, old_height, old_width = shape[-3:]
+    old_height, old_width = image.shape[-2:]
     new_height, new_width = _compute_resized_output_size((old_height, old_width), size=size, max_size=max_size)
 
     if (new_height, new_width) == (old_height, old_width):
         return image
-    elif numel > 0:
-        image = image.reshape(-1, num_channels, old_height, old_width)
 
-        dtype = image.dtype
-        acceptable_dtypes = [torch.float32, torch.float64]
-        if interpolation == InterpolationMode.NEAREST or interpolation == InterpolationMode.NEAREST_EXACT:
-            # uint8 dtype can be included for cpu and cuda input if nearest mode
+    if image.ndim > 2:
+        num_channels = image.shape[-3]
+        output_shape = image.shape[:-3] + (num_channels, new_height, new_width)
+    else:
+        num_channels = 1
+        output_shape = [new_height, new_width]
+
+    numel = image.numel()
+    if numel == 0:
+        return image.reshape(output_shape)
```

Comment:
This is the only new functionality. Since we now compute the `output_shape` upfront, we can make the `numel == 0` case a shortcut and return early. Due to this, I removed the `if / else` branching below and thus could remove one level of indent. Unfortunately, `git` is not picking up on that and thus this diff looks larger than it is. There is no functional change to the core of `F.resize_image`.